### PR TITLE
NEX-1335 add OWASP Dependency Check Plugin

### PR DESCRIPTION
**Pull request by mike-unitskyi**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -49,6 +53,24 @@
...
+                    <version>1.3.1</version>
+                    <executions>
+                        <execution>
+                            <goals>
+                                <goal>check</goal>
+                            </goals>
+                            <configuration>
+                                <failBuildOnCVSS>6</failBuildOnCVSS>
+                                <suppressionFile>
+                                    ${session.executionRootDirectory}/owasp-suppression.xml
```
Comment (ohhatiya) [inline pom.xml]: Why `session.executionRootDirectory` instead of `project.basedir`?

File: owasp-suppression.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<suppressions
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns="https://www.owasp.org/index.php/OWASP_Dependency_Check_Suppression"
+        xsi:schemaLocation="https://www.owasp.org/index.php/OWASP_Dependency_Check_Suppression suppression.xsd"
+>
+    <suppress>
+        <notes><![CDATA[
+   file name: jackson-datatype-jdk7-2.6.7.jar
```
Comment (ohhatiya) [inline owasp-suppression.xml]: Out of curiosity, why do we need these suppressions? Can we just update to a newer jackson?

File: pom.xml (hunk 1 above)
Comment (mike-unitskyi) [reply pom.xml]: actually they are the same in this context. Followed https://bits.bazaarvoice.com/confluence/display/SEC/How+to+Integrate+Security+Library+Dependency+Checks+Into+Your+Java+Project doc.

File: owasp-suppression.xml (hunk 2 above)
Comment (mike-unitskyi) [reply owasp-suppression.xml]: we use latest jackson version(2.9.4).
jackson-datatype-jdk7 was [deprecated](https://github.com/FasterXML/jackson-datatype-jdk7) and latest version is `2.6.7`. Now `jackson-datatype-jdk7` is part of https://github.com/FasterXML/jackson-databind so we can even exclude `jackson-datatype-jdk7`.
```bash
mvn dependency:tree | grep jackson-datatype-jdk7 -C 3
[INFO] |  +- io.dropwizard:dropwizard-jackson:jar:0.9.0:compile
[INFO] |  |  +- com.fasterxml.jackson.core:jackson-core:jar:2.9.4:compile
[INFO] |  |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.9.4:compile
[INFO] |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-jdk7:jar:2.6.3:compile
[INFO] |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-guava:jar:2.9.4:compile
[INFO] |  |  +- com.fasterxml.jackson.module:jackson-module-afterburner:jar:2.9.4:compile
[INFO] |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-joda:jar:2.9.4:compile
```

**Comment by mike-unitskyi** [comment]
@ohhatiya , please review again

File: owasp-suppression.xml (hunk 2 above)
Comment (ohhatiya) [reply owasp-suppression.xml]: Oh, I see. It's a transitive dependency of Dropwizard. And I'm guessing excluding that dependency in our pom and getting the newer databind module breaks Dropwizard.

File: pom.xml (hunk 1 above)
Comment (ohhatiya) [reply pom.xml]: Hmm. I think I like using `project.basedir` better. In practice they'll typically be the same, but it seems wrong to potentially pick up some other random owasp-suppression.xml if maven is executed from a different directory. In fact, I think I'm going to update the example in Confluence as well.

Comment (mike-unitskyi) [reply pom.xml]: `project.basedir` doesn't work for multimodule projects

File: owasp-suppression.xml (hunk 2 above)
Comment (mike-unitskyi) [reply owasp-suppression.xml]: it could break, but seems like we don't use json serialization for jdk7 datatypes

**Pull request merged**